packages/react/src/webview/webViewManager.ts

```diff
@@ -0,0 +1,444 @@
+    });
+
+    return true;
+  }
+
+  /**
+   * @description
+   * Validates the target origin for the web view.
+   * If no origin is specified, it defaults to the current window's origin.
+   * If the specified origin is invalid, it also defaults to the current window's origin.
+   * @param origin The target origin to validate.
+   * @returns
+   */
+  private validateTargetOrigin(origin: string): string {
+    if (!origin) {
+      console.warn('No target origin specified, using current origin');
+      return window.location.origin;
+    }
+
+    try {
+      new URL(origin); // Validate it's a proper URL
+      return origin;
+    } catch {
+      console.warn('Invalid origin specified, using current origin');
+      this.debug &&
+        console.warn('[WebViewManager] Invalid origin, using current origin');
+      return window.location.origin;
+    }
+  }
+
+  /**
+   * @description
+   * Sets the target origin for the web view.
+   * If no origin is specified, it defaults to the current window's origin.
+   * If the specified origin is invalid, it also defaults to the current window's origin.
+   * @returns targetOrigin on which the web view is opened
+   */
+  private get targetOrigin(): string {
+    return this.validateTargetOrigin(
+      TARGET_ORIGIN_RESPONSE ??
+        this.allowedOrigins?.[0] ??
+        window.location.origin,
+    );
+  }
+
+  /**
+   * @description
+   * Handles incoming messages from the web view.
+   * @param method The method name to handle.
+   * @param handler The handler function to call when the method is invoked.
+   * @returns void
+   * @example
+   * webViewManager.onRequest('exampleMethod', async (data) => {
+   *   return { success: true };
+   * });
+   */
+  public onRequest(method: string, handler: (data: any) => Promise<any>): void {
+    this.requestHandlers.set(method, handler);
+  }
+
+  /**
+   * @description
+   * Sends a request to the web view and returns a promise that resolves with the response.
+   * @param method The method name to call.
+   * @param data The data to send with the request.
+   * @returns A promise that resolves with the response data.
+   * @example
+   * webViewManager.sendRequest('exampleMethod', { key: 'value' })
+   *   .then((response) => {
+   *     console.log('Response:', response);
+   *   })
+   *   .catch((error) => {
+   *     console.error('Error:', error);
+   *   });
+   * @throws Error if the web view is not open or if the request times out.
+   * @throws Error if the request fails.
+   * @throws Error if the web view is not available.
+   * @throws Error if the request times out.
+   * @throws Error if the request fails.
+   */
+  public sendRequest(method: string, data: any): Promise<any> {
+    return new Promise((resolve, reject) => {
+      if (!this.isWebViewOpen()) {
+        this.debug &&
+          console.warn(
+            '[WebViewManager] WebView not available for sending request',
+          );
```

Comment:
eslint: Expected an assignment or function call and instead saw an expression.

Reply:
@SrijanCoinDCX it is in multiple lines of code across codebase. please fix all eslint errors

Reply:
++ @karansinglacoindcx similar errors in RN as well